packages/patternfly-4/react-core/src/components/Alert/Alert.tsx

```diff
@@ -79,10 +98,11 @@ const Alert: React.FunctionComponent<AlertProps & InjectedOuiaProps> = ({
       })}
     >
       <AlertIcon variant={variant} />
-      <h4 className={css(styles.alertTitle)}>{readerTitle}</h4>
+      <h4 className={css(styles.alertTitle)}>{getHeadingContent}</h4>
       {children && <div className={css(styles.alertDescription)}>{children}</div>}
-      {action && (
-        <div className={css(styles.alertAction)}>{React.cloneElement(action as any, { title, variantLabel })}</div>
+      {action && typeof action === 'object' && <div className={css(styles.alertAction)}>{action}</div>}
+      {action && typeof action === 'function' && (
+        <div className={css(styles.alertAction)}>{action(title, variantLabel)}</div>
```

Comment:
Can you pass the `title` and `variantLabel` as object? Also what are your thoughts on ignoring altogether the `div` and pass the alertAction className to action callback?
```JS
{action && typeof action === 'function' && action({ title, variantLabel, className: css(styles.alertAction) })}
```

This can be helpfull for consumers to write whatever they want as action, but it has drawbacks that they have to use the className from the action callback props.

Reply:
@karelhala split this out into another PR so we can get the bug fixed without breaking change https://github.com/patternfly/patternfly-react/pull/3822

You can achieve something similar to what you describe without the div by using a fragment, but I see what you mean. We can definitely explore those changes over in #3822 if that works for you

Reply:
I think that this is great first step! Let's iterate on it, I think it is good to keep it this way and if any need for custom component as a wrapper arise we can address it later on.